include/ratelimit_manager.h

```diff
@@ -222,10 +224,16 @@ class RateLimitManager
         // Helper function to get ban key for DB store from ID
         std::string get_ban_key(const uint32_t id);
 
-        // Helper function to 
+        // Base timestamp
+        time_t base_timestamp = 0;
+
+        // Helper function to get current timestamp
+        time_t get_current_time();
 
         // Singleton instance
         inline static RateLimitManager *instance;
 
+        friend class RateLimitManagerTest;
```

Comment:
Not a fan of friend class. Let's have a public `_set_current_time` function that the test uses (name begins with `_` to indicate internal use).  With this, we also don't need forward declaration of `RateLimitManagerTest` above.